public/app/features/admin/UserListAdminPage.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux';
+import { NavModel } from '@grafana/data';
+import { Pagination, Forms, ActionBar, ActionBarSpacing, Tooltip } from '@grafana/ui';
+import { StoreState, UserDTO } from '../../types';
+import Page from 'app/core/components/Page/Page';
+import { getNavModel } from '../../core/selectors/navModel';
+import { fetchUsers, changeQuery, changePage } from './state/actions';
+import { TagBadge } from 'app/core/components/TagFilter/TagBadge';
+
+interface OwnProps {}
+
+interface ConnectedProps {
+  navModel: NavModel;
+  users: UserDTO[];
+  query: string;
+  showPaging: boolean;
+  totalPages: number;
+  page: number;
+}
+
+interface DispatchProps {
+  fetchUsers: typeof fetchUsers;
+  changeQuery: typeof changeQuery;
+  changePage: typeof changePage;
+}
+
+type Props = OwnProps & ConnectedProps & DispatchProps;
+
+const UserListAdminPageUnConnected: React.FC<Props> = props => {
+  useEffect(() => {
+    props.fetchUsers();
```

Comment:
I guess this is an async action? I think we have some nice async hooks supplied with an external lib that I can't remember the name of right now.  ping @dprokop

Reply:
Yes, just want it to be run at the same "cycle" as component did mount.

Reply:
Thanks @hugohaggmark for mentioning about this. @mckn I think you can use useAsync from react-use (https://github.com/streamich/react-use/blob/master/src/useAsync.ts)

Reply:
hmm, since it won't return any value via a promise I don't think the `useAsync` is applicable in this scenario.

Reply:
Why not? Not sure I follow what return value has to do here. The nice thing about useAsync is that you can render some loading indication while the list is being loaded

Reply:
Yeah - saw that. Looks really nice! But since this is a thunks I already have all the benefits of handling this inside the thunk + redux. If this wasn't a connected component I totally see the benefit of using the `useAsync` hooks but in this case I think I would prefer to keep it consistent. 

If that makes sense? But a question that I don't have resolved is what should happen if the async code fails? How will I display a potential error? Do we have any pattern for that?

Reply:
> If this wasn't a connected component I totally see the benefit of using the useAsync hooks but in this case I think I would prefer to keep it consistent.

👍 

> But a question that I don't have resolved is what should happen if the async code fails? How will I display a potential error? Do we have any pattern for that?

We actually don't have an "inline error" pattern for such cases, but the backendSrv error will be handled with global app notification (red box in top right corner).

Reply:
Aha, awesome! I need to have a look at that one. Thanks! 🙏